jerry-libc/jerry-libc.c

```diff
@@ -158,41 +158,15 @@ CALL_PRAGMA (GCC diagnostic pop)
 int
 strcmp (const char *s1, const char *s2)
 {
-  size_t i;
-  if (s1 == NULL)
+  for (; *s1 == *s2; s1++, s2++)
   {
-    if (s2 != NULL)
-    {
-      return -1;
-    }
-    else
+    if (*s1 == '\0')
     {
       return 0;
     }
   }
-  if (s2 == NULL)
-  {
-    return 1;
-  }
-
-  for (i = 0; s1[i]; i++)
-  {
-    if (s1[i] > s2[i])
-    {
-      return 1;
-    }
-    else if (s1[i] < s2[i])
-    {
-      return -1;
-    }
-  }
-
-  if (s2[i])
-  {
-    return -1;
-  }
 
-  return 0;
+  return (*(const unsigned char *)s1 - *(const unsigned char *)s2);
```

Comment:
Not necessary, strcmp() does not need to return -1 or 1 specifically. The important part is the sign of the return value.